mm/mm_heap/mm_mallinfo.c

```diff
@@ -105,9 +106,10 @@ int mm_mallinfo(FAR struct mm_heap_s *heap, FAR struct mallinfo *info)
             }
           else
             {
-              FAR struct mm_freenode_s *fnode;
+#ifdef CONFIG_DEBUG_ASSERTIONS
+              FAR struct mm_freenode_s *fnode = (FAR void *)node;
+#endif
               DEBUGASSERT(node->size >= SIZEOF_MM_FREENODE);
-              fnode = (FAR void *)node;
               DEBUGASSERT(fnode->blink->flink == fnode);
```

Comment:
This assertion can't be right.  If line 110 is removed, then fnode has no value.

Reply:
You mean it won't have a value for DEBUGASSERT?
But DEBBUGASSERT is only active when assertions are enabled.

Reply:
Then don't you get a warning about fnode being unused?

Reply:
Perhaps including all of lines 108 though 117 in #ifdef CONFIG_DEBUG_ASSERTIONS would eliminate warnings about the unused variable. when CONFIG_DEBUG_ASSERTIONS is not defined.

Reply:
This is the original warning (among 2 others) that this PR fixes.
When assertions are disabled `fnode` and `prev` were not used.

Reply:
Oh, I just noticed that fnodes is doubly initialized.  That is why you can remove the second initialization.  Sorry, I didn't follow that corrected.

This should still cause a warning about an unused variable and still should be enclosed in conditional logic.